index.js

```diff
@@ -72,6 +72,12 @@ class ServerlessWebpack {
               'compile',
             ],
           },
+          "watch-compile": {
```

Comment:
You should move that as subcommand to the compile command (as entrypoint):
```js
compile:
  ...
  commands: {
    watch: {
      type: 'entrypoint',
      lifecycleEvents: [
        'compile'
      ]
    }
  }
```
Then it is more universal and better located semantically.

Additionally add a hook to define the new hookable event in `this.hooks` like this:
```js
this.hooks = [
  ...
  'webpack:compile:watch:compile': () => BbPromise.resolve();
};
```
so that it is owned by the plugin and can be addressed with
`after:webpack:compile:watch:compile` and the symmetric before.